presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -370,4 +373,17 @@ public void testCorrelatedScalarAggregationRewriteToLeftOuterJoin()
                                                                                 project(ImmutableMap.of("NON_NULL", expression("true")),
                                                                                         node(ValuesNode.class)))))))))));
     }
+
+    @Test
+    public void testPruneCountAggregationOverScalar()
```

Comment:
Please also test following queries:

SELECT count(*) from (select sum(orderkey) from orders GROUP  BY customerkey)
SELECT EXISTS(SELECT sum(orderkey) from orders)

With the rule you are adding, the second query should also be optimized to VALUES